.travis.yml

```diff
@@ -0,0 +1,14 @@
+os: osx
+osx_image: xcode11.3
+
+before_install:
+  - brew update
+  - brew tap osx-cross/avr
+  - brew tap osx-cross/arm
+  - brew install scons git doxygen boost gcc avr-gcc arm-gcc-bin cmake || true
+  - brew upgrade boost gcc git || true
+  - export PATH=/Users/travis/Library/Python/3.7/bin:$PATH
+  - pip3 install --user modm
+
+script:
+  - (cd examples && ../tools/scripts/examples_compile.py linux)
```

Comment:
Don't we also want to compile at least the hosted darwin unit tests and a few examples of AVR and STM32? Not all, it'll take too long.

Reply:
On my host ...

```
➜  modm-str-ci_macos git:(feature/ci_macos) (cd test && make compile-arduino-nano)
lbuild -c config/arduino-nano.xml  -p ../build/generated-unittest/arduino-nano build --no-log
[WARNING] user.modm-test.test.platform.gpio: GPIO test has been hardcoded to a Nucleo-64 board. When porting make sure this test does not damage your board!

ERROR: In Module 'modm:platform:core':
In file '../src/modm/platform/core/avr/module.lb:65' in function 'build':
        env.template("delay_impl.hpp.in")

Error in template '../src/modm/platform/core/avr/delay_impl.hpp.in:47':
	}) : ({
%% if f_cpu <= 6e6
		constexpr int32_t cycles = (F_CPU / 1e6);
		for (int16_t _us = us; _us > 0; _us -= cycles)

TemplateSyntaxError: expected token 'end of statement block', got 'e6'
make: *** [compile-arduino-nano] Error 1
```

I enabled some, but we need to fix this first ...

Reply:
Can't reproduce on my local machine and you seen to have fixed it anyways?